src/extensions/core/load3d.ts

```diff
@@ -0,0 +1,735 @@
+        }
+
+        this.renderer.render(this.scene, this.activeCamera)
+
+        const imageData = this.renderer.domElement.toDataURL('image/png')
+
+        this.renderer.setSize(originalWidth, originalHeight)
+        this.handleResize()
+
+        resolve(imageData)
+      } catch (error) {
+        reject(error)
+      }
+    })
+  }
+
+  setViewPosition(position: 'front' | 'top' | 'right' | 'isometric') {
+    const box = new THREE.Box3()
+    let center = new THREE.Vector3()
+    let size = new THREE.Vector3()
+
+    if (this.currentModel) {
+      box.setFromObject(this.currentModel)
+      box.getCenter(center)
+      box.getSize(size)
+    }
+
+    const maxDim = Math.max(size.x, size.y, size.z)
+    const distance = maxDim * 2
+
+    switch (position) {
+      case 'front':
+        this.activeCamera.position.set(0, 0, distance)
+        break
+      case 'top':
+        this.activeCamera.position.set(0, distance, 0)
+        break
+      case 'right':
+        this.activeCamera.position.set(distance, 0, 0)
+        break
+      case 'isometric':
+        this.activeCamera.position.set(distance, distance, distance)
+        break
+    }
+
+    this.activeCamera.lookAt(center)
+    this.controls.target.copy(center)
+    this.controls.update()
+  }
+
+  setBackgroundColor(color: string) {
+    this.renderer.setClearColor(new THREE.Color(color))
+    this.renderer.render(this.scene, this.activeCamera)
+  }
+}
+
+function splitFilePath(path: string): [string, string] {
+  const folder_separator = path.lastIndexOf('/')
+  if (folder_separator === -1) {
+    return ['', path]
+  }
+  return [
+    path.substring(0, folder_separator),
+    path.substring(folder_separator + 1)
+  ]
+}
+
+function getResourceURL(
+  subfolder: string,
+  filename: string,
+  type: string = 'input'
+): string {
+  const params = [
+    'filename=' + encodeURIComponent(filename),
+    'type=' + type,
+    'subfolder=' + subfolder,
+    app.getRandParam().substring(1)
+  ].join('&')
+
+  return `/view?${params}`
+}
+
+app.registerExtension({
+  name: 'Comfy.Load3D',
+
+  getCustomWidgets(app) {
+    return {
+      LOAD_3D(node, inputName) {
+        let load3dNodes = app.graph._nodes.filter((wi) => wi.type == 'LOAD_3D')
+
+        node.name = `load3d_${load3dNodes.length}`
+
+        const containerWrapper = document.createElement('div')
+
+        containerWrapper.id = `load-3d-wrapper-${inputName.toLowerCase()}`
+        containerWrapper.classList.add('comfyui-load-3d-wrapper')
+
+        const container = document.createElement('div')
+        container.id = `load-3d-${inputName.toLowerCase()}`
+        container.classList.add('comfyui-load-3d')
+        containerWrapper.appendChild(container)
+
+        node.load3d = new Load3d(node, container)
+
+        node.onResize = function () {
+          let [w, h] = this.size
+          if (w <= 300) w = 300
+          if (h <= 300) h = 300
+          this.size = [w, h]
+
+          if (this.load3d) {
```

Comment:
`this` ref is ambiguous here. Please replace it with `node` if you are trying to bind something to the LGraphNode object. With previous comment in mind, here you probably want to read from widget object instead.

Reply:
fixed